Added additional testing for error handling in main.py and tree_processor.py

devin201o opened the pull request:
<!-- 
Thank you for contributing! Please fill out this template to help us review your PR.
-->

## 📝 Description

> Please include a summary of the change and which issue is fixed.  
> Include relevant motivation and context.  Test that permission errors are caught and converted
> List any dependencies that are required for this change.

**Closes:** #121 

Added additional testing for `main.py` and `tree_processor.py` specifically targeting error handling:
- `tree_processor.py`
   - Tests that passing None as root raises error
   - Tests that passing a non object argument raises typeError
   - tests that problematic nodes wont stop the tree from being processed
 - `main.py`
   - test that empty filepaths raises valueErrors
   - Test that path resolution errors are caught and converted to ValueError
   - Test that permission errors are caught and converted

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [ ] ♻️ Refactoring
- [ ] ⚡ Performance improvement

---

## 🧪 Testing

> Please describe how you tested this PR (both manually and with tests).  
> Provide instructions so we can reproduce.

All tests run and pass within the Docker container 

- `tree_processor.py`
   - Tests that passing None as root raises error
   - Tests that passing a non object argument raises typeError
   - tests that problematic nodes wont stop the tree from being processed
 - `main.py`
   - test that empty filepaths raises valueErrors
   - Test that path resolution errors are caught and converted to ValueError
   - Test that permission errors are caught and convertedz
---

## ✓ Checklist

- [ ] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [x] 💬 I have commented my code where needed
- [ ] 📖 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [ ] 🔗 Any dependent changes have been merged and published in downstream modules
- [ ] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## 📸 Screenshots

> If applicable, add screenshots to help explain your changes

<details>
<summary>Click to expand screenshots</summary>

<!-- Add your screenshots here -->

</details>

antran-17 reviewed (approved):
Included many test cases for error handling and all tests passed on Docker container, great job!

devin201o commented:
> Good job on the mocked testing, and added edge case testing. Just to confirm line 216 is an intentional typo yes?

Good catch, this is an unintentional typo, but it should not affect the testing. I just fixed the typo in the latest commit, thanks!

maddydeg reviewed (approved):
This implementation is well done and all tests are passing locally. Requested changes from Sara have been made, great work on this one!! I am approving and we should be good to get this one merged.
